### DISCUSSION TITLE
Replace 'dependabot' with 'Renovate'

### DISCUSSION THREAD

**Discussion by hazendaz** [oshi/oshi#2157] [General]
Renovate is a much better tool for managing dependencies.  Rather than a yml file we have to maintain, it creates a nice dependency dashboard issue that keeps track of what we don't want via close and further all dependencies discovered in the project.  Unlike dependabot it further knows about far more and now has much better sponsorship.  I'll do the work to switch over but wanted to raise it here to get thoughts before doing so as it will cause a bit of noise during switch over.

A sample dashboard can be found here which further will get you to the docs [renovate](https://github.com/mybatis/guice/issues/466)

**Comment by dbwiddis** [comment]
AFAIK the yml file doesn't take much maintenance, and a brief look at Renovate docs showed you still need some sort of JSON config somewhere (perhaps on their site?).  I can see the value for a project which doesn't regularly update every single dependency, to keep track of what's still on a previous version, but in our case we're almost always up to date with everything.

Not saying no, just saying after a brief glance I'm not yet seeing much difference.  But I haven't looked very far. 

Will be offline for a few days so don't interpret lack of further replies as anything other than me not having read them yet. :)

I'm ok with switching in any case if you're doing the work, we can always switch back if I hate it.

**Reply by dbwiddis** [reply]
I do see that Renovate has batch updates, which I like, although that also makes it harder to pin down bugs that occasionally happen.

**Reply by hazendaz** [reply]
json is one way of doing things but with its default integration with github, it opens an issue that just lives forever that it tracks that on and makes it easy to reconsider items so no file to maintain manually.  I've been using this at my day job for a couple years and had championed its usage.  Mainly because we use bitbucket but its gotten so much better than dependabot that it shows.  I've turned it on a bunch of places here in last week and its further discovered a number of missed opportunities and even resolved some long standing vulnerabilities on one repo I maintain.  I can go ahead and make the change.  Its basically just removal of the dependabot file once we stabilize on it.  We could keep both on for short period of time just to see differences in how they behave.  I have that one one repo for more than a year at this point but kind of made up my mind I'm done with dependabot.

**Reply by dbwiddis** [reply]
Make it so, number one!

**Comment by hazendaz** [comment]
renovate added.  as noted, was not much to do.  It has one ignore.  If you look at the issues, you will see long running dependency dashboard.  It has the one ignore we had for dependabot now.  If ever need to run that ignored one, just click the box and it will do it.  Lot easier than dependabot in so many ways.  We just need to click rather than at dependabot and it keeps data in one spot that way.

**Reply by hazendaz** [reply]
to remove dependabot, we only need to delete the yaml file.

**Comment by dbwiddis** [comment]
So I like the dashboard.  The pull requests have less value as they don't link to the changed commits like dependabot did.  I like to read those to identify new features I might want to employ (or in the recent case of junit, changed behavior!). I can probably still find those manually, but it is something I miss.

Can we turn on batch merging, at least for patch versions?  I might even be happy with batch on all maven dependencies, leaving manual updates for java dependencies, if that's easy to set up.

Also it looks like we can move the json config file into the .github directory. Since getting rid of a config file was an advertised benefit, moving it off the main page would be nice :)

**Comment by dbwiddis** [comment]
There's also a big ugly warning on the dashboard.

> ### ⚠ Dependency Lookup Warnings ⚠
> 
> -   Renovate failed to look up the following dependencies: `org.eclipse.m2e:lifecycle-mapping`.

Research indicates the plugin is not at Central, but at the [Gluu](https://ox.gluu.org/maven/) repository.  But it also doesn't really exist; it's in its own profile which is only activated during eclipse builds where the `m2e.version` property is set, and m2e knows what to do with the config:
https://github.com/oshi/oshi/blob/c261bf88088d1b4f621d9fa3eda3685989782b97/pom.xml#L1035-L1050

Do we really need this config?  I build in Eclipse but I (almost) always run maven from the command line.  If we keep it is there a way to get this ugly warning off the dashboard?

**Comment by hazendaz** [comment]
I suspect we can either add an entry I renovate file to ignore the m2e marker.  It is needed or imports of new project to eclipse will throw lots of warnings unless plugins have been updated.  What you could try is to delete that section, delete from eclipse including settings classpath and properties.  Then import and see what happens.  I can take a look at the ignore route.

On pull requests, what doesn't link?  Label maybe?  That's simply enough to address.

Renovate is extremely active releasing 3 times a day or more most days.  They don't incorrectly run on forks either unless told to do so and even then can be stopped.  For me that was and is dependabots downfall.  I had to clear probably 500 old prs on forks yesterday alone as I dropped dependabot everywhere.  It also was not a very smart tool.

Renovate is far more advanced, we may just need to configure it beyond the defaults.  It's highly extensible.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by dbwiddis** [reply]
> On pull requests, what doesn't link? 

Look at any dependabot PR.  Under the top line (bumps x to y) there is a section labeled "commits". 

Sometimes if a repo is configured in a certain way (not sure how but I see it often enough) it also links to a change log. 

If I wanted to see *why* the javadoc plug-in got a version bump I would (and did) have to go to the plug-in site and hunt down the release notes.

**Reply by dbwiddis** [reply]
Here's an example of one with both release notes and commits.
https://github.com/oshi/oshi/pull/2148

**Reply by dbwiddis** [reply]
>  It is needed or imports of new project to eclipse will throw lots of warnings unless plugins have been updated.  

I'm not sure that's the case. The way we're using it, I think it just adds a configuration to ignore some plugins for one or more of the m2e maven options:
![image](https://user-images.githubusercontent.com/9291703/184555214-c6c0329f-5aa9-442c-9a80-6fa56f3218e5.png)

As far as our config, it appears to do three things:
 - ignore the license format step (which we don't execute in our current config anyway because it's slow)
 - skip jacoco, which could save a few seconds of the build
 - skip the enforcer plugin enforce goal, which doesn't take long to run anyway

So I think it's probably completely safe to remove.

**Reply by dbwiddis** [reply]
Rather than ignoring we should just add the gluu repository as a possible repo for lookups.  The plugin version is obviously very old but seems to have been recently published at Gluu, possibly by annoyed Renovate users :)

**Comment by hazendaz** [comment]
Renovate does show release notes if they were done appropriately.  Maven is probably a bad use case as their git repos are mirrors and they use jira.  Would need to see an example from prior maven to see if that was any different than link to their source.  I'll look around on that.  It's also quite possible when the cut it that they may not have been available yet as well.  This happened faster than their voting.  If you peak at their mailing list as least for javadoc. It was announced out but still didn't get all votes until very shortly after.  Renovate send PRs almost immediately mainly because I boarded at roughly same time.

**Reply by dbwiddis** [reply]
> Renovate does show release notes if they were done appropriately. 

By whose definition of "appropriately"? 😁 

I agree that the "release notes" are probably similar and sufficient, and if it's super-important I could hunt down the diff in commits (although that's probably easily put in a link in a future Renovate feature similar to how dependabot does it.)

Anyway, I'm happy to tweak Renovate... I'd love to reduce the number of version bump PRs for build tools.

**Comment by hazendaz** [comment]
Here is a better example with release notes.  https://github.com/hazendaz/dependency-tracker/pull/2395

**Comment by mprins** [comment]
the links to repositories that the renovate bot generates route through `togithub.com` links which appears to be some sort of tracking service that has unknown registrants (they use https://withheldforprivacy.com/ )

**Comment by dbwiddis** [comment]
Something must be wrong with my configuration.  We got two updates (released simultaneously) and they showed on the dashboard but I had to manually "click the check box" to trigger the PR that just happened.  I thought it would happen daily (between midnight and 2am UTC).  I apparently need to read the docs...

**Reply by hazendaz** [reply]
renovate config can be tricky, at least there was a fall back method :)

**Comment by dbwiddis** [comment]
I think I used a 'preset' incorrectly. Oh, well. I think I fixed it.

Please don't merge the PR, I'm testing automerge. :)
